tools/chpl-language-server/test/type-inlays.py

```diff
@@ -0,0 +1,491 @@
+"""
+Test that type inlays show up properly, and that they can function as
+clickable links. Requires `--resolver`.
+"""
+
+import sys
+
+from lsprotocol.types import ClientCapabilities
+from lsprotocol.types import HoverParams
+from lsprotocol.types import InitializeParams
+import pytest
+import pytest_lsp
+from pytest_lsp import ClientServerConfig, LanguageClient
+
+from util.utils import *
+from util.config import CLS_PATH
+
+
+@pytest_lsp.fixture(
+    config=ClientServerConfig(
+        server_command=[
+            sys.executable,
+            CLS_PATH(),
+            "--resolver",
+            "--type-inlays",
+            "--no-literal-arg-inlays",
+            "--no-param-inlays",
+            "--end-markers=none",
+        ],
+        client_factory=get_base_client,
+    )
+)
+async def client(lsp_client: LanguageClient):
+    # Setup
+    params = InitializeParams(capabilities=ClientCapabilities())
+    await lsp_client.initialize_session(params)
+
+    yield
+
+    # Teardown
+    await lsp_client.shutdown_session()
+
+
+@pytest.mark.asyncio
+async def test_type_inlays_prim(client: LanguageClient):
+    """
+    Ensure that type inlays are shown for primitive types.
+    """
+
+    file = """
+            var a = 10;
+            var b: int = 10;
+            var c = a:int(32);
+            var d = 0.0;
+
+            var e = "hello";
+            var f = true;
```

Comment:
Inspired by your testing of type inlays for split-init generic types, consider doing split-init here. it should work, I've seen it work with params.